File: src/main/java/com/example/kyrgyzstancentralmedicalcard/services/DiagnosisService.java
package com.example.kyrgyzstancentralmedicalcard.services;


import com.example.kyrgyzstancentralmedicalcard.entity.Diagnosis;

import java.util.List;

public interface DiagnosisService {
    List<Diagnosis> getAllDiagnosis();

    Diagnosis createDiagnosis(Diagnosis diagnosis, Long id);

    Diagnosis getById(Long id);

    List<Diagnosis> getByName(String name);

    Diagnosis updateDiagnosis(Diagnosis diagnosis, Long id);

    List<Diagnosis> getAllDiagnosesByUserId(Long userId); // Добавленный метод
}

File: frontend-react/src/services/diagnosisService.js
import api from './api';

const createDiagnosis = (userId, data) => api.post(`/diagnosis/create-diagnosis/${userId}`, data);
const getAllDiagnosesByUserId = (userId) => api.get(`/diagnosis/get-all-by-user/${userId}`);

const diagnosisService = { createDiagnosis, getAllDiagnosesByUserId };
export default diagnosisService;
